proto/redpanda/api/console/v1alpha1/security.proto

```diff
@@ -0,0 +1,751 @@
+  }
+
+  // The role name.
+  string role = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Optional filter.
+  optional Filter filter = 2;
+
+  // Page size.
+  int32 page_size = 3 [
+    (buf.validate.field).int32 = {
+      gte: -1,
+      lte: 1000
+    },
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "Limit the paginated response to a number of items. Defaults to 100. Use -1 to disable pagination.",
+      minimum: -1,
+      maximum: 1000
+    }
+  ];
+
+  // Value of the next_page_token field returned by the previous response.
+  // If not provided, the system assumes the first page is requested.
+  string page_token = 4;
+}
+
+// ListRoleMembersResponse is the response for ListRoleMembers.
+message ListRoleMembersResponse {
+  // Members assigned to the role.
+  repeated RoleMembership members = 1;
+}
+
+// RoleMembership is the role membership.
+message RoleMembership {
+  // The name of the principal assigned to the role.
+  string name = 1;
+
+  // The principal type.
+  Security.PrincipalType principal_type = 2;
+}
+
+// UpdateRoleMembershipRequest is the request to UpdateRoleMembership.
+message UpdateRoleMembershipRequest {
+  // The role name.
+  string role = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Create the role if it doesn't already exist.
+  // If the role is created in this way, the “add” list will be respected, but the “remove” list will be ignored.
+  optional bool create = 2;
+
+  // Members to assign to the role.
+  repeated RoleMembership add = 3;
+
+  // Members to remove from the role.
+  repeated RoleMembership remove = 4;
+}
+
+// UpdateRoleMembershipResponse is the response for UpdateRoleMembership.
+message UpdateRoleMembershipResponse {
+  // The role name.
+  string role = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.min_len = 1,
+    (buf.validate.field).string.max_len = 128,
+    (buf.validate.field).string.pattern = "^[a-zA-Z0-9-_]+$"
+  ];
+
+  // Members assigned to the role.
+  repeated RoleMembership added = 2;
+
+  // Members removed from the role.
+  repeated RoleMembership removed = 3;
+}
+
+// ListUserRolesRequest is the request for ListUserRoles.
+message ListUserRolesRequest {
+  // Filter options.
+  message Filter {
+    // Filter results to only roles named with the prefix.
+    string name_prefix = 1 [
+      (buf.validate.field).string.max_len = 128,
+      (buf.validate.field).string.pattern = "^([a-zA-Z0-9-_]*)$"
+    ];
+  }
```

Comment:
I have not looked at the Core RFC for a while, so I'm actually not sure what this endpoint does. I was unsure whether it's:

1. Return all users that belong to role x
2. Return all roles that are bound to a user x

Also wondering whether the Filter is actually optional or required here. Is the term `user` as part of the request/response name apropriate here or is it a principal?

Reply:
There is a comment on the actual RPC definition.
```
// ListUserRoles lists all the authenticated user's roles based on optional filter.
```

Maybe the RPC can be renamed to be clearer?

The filter is for filtering the list roles in the response.

Reply:
> lists all the **authenticated** user's

Authenticated user is the Console superuser here? That endpoint wouldn't help us in Console as of today? Why would we expose an endpoint that returns the Console user's role?

Reply:
I remove `ListUserRoles` rpc.